webui/react/src/utils/browser.ts

```diff
@@ -16,3 +20,66 @@ export const getCookie = (name: string): string | null => {
   const value = document.cookie.replace(regex, '$1');
   return value ? value : null;
 };
+
+const downloadBlob = (filename: string, data: Blob): void => {
+  const url = window.URL.createObjectURL(data);
+  const element = document.createElement('a');
+  element.setAttribute('download', filename);
+  element.style.display = 'none';
+  element.href = url;
+  document.body.appendChild(element);
+  element.click();
+  window.URL.revokeObjectURL(url);
+  document.body.removeChild(element);
+};
+
+// Different JS engines impose different maximum string lenghts thus for downloading
+// large text we split it into different parts and then stich them together as a Blob
+export const downloadText = (filename: string, parts: BlobPart[]): void => {
+  const data = new Blob(parts, { type: 'text/plain' });
+  downloadBlob(filename, data);
+};
+
+export const downloadTrialLogs = async (trialId: number): Promise<void> => {
+  // string concatnation is fater than array.join but is also more memory friendly?
+  // https://jsben.ch/OJ3vo
+  // characters are utf16 encoded. v8 has different internal representations.
+  const MAX_PART_SIZE = 128 * Math.pow(2, 20); // 128m * CHAR_SIZE
+  const parts: BlobPart[] = [];
+  let downloadStringBuffer = '';
+  await consumeStream<DetSwagger.V1TrialLogsResponse>(
+    DetSwagger.ExperimentsApiFetchParamCreator().determinedTrialLogs(trialId),
+    (ev) => {
+      if (downloadStringBuffer.length > MAX_PART_SIZE) {
+        parts.push(downloadStringBuffer);
+        downloadStringBuffer = '';
+      }
+      downloadStringBuffer += ev.message;
+    },
+  );
+  const trial = await getTrialDetails({ id: trialId });
+  downloadText(`experiment_${trial.experimentId}_trial_${trialId}_logs.txt`, parts);
+};
+
+const generateLogStringBuffer = (count: number, avgLength: number): string => {
+  const msg = new Array(avgLength).fill('a').join('') + '\n';
+  let stringBuffer = '';
+  for (let i=0; i<count; i++) {
+    stringBuffer += (i + msg);
+  }
+  return stringBuffer;
+};
+
+// TEMP TODO: remove me
```

Comment:
probably remove the test/benchmark code?

Reply:
**thought:** How about updating the `LogViewer.stories.tsx` to have a large view/download example?

Reply:
that's a good idea.

Reply:
I added this with configurable log size

Reply:
beautiful